### PR TITLE
Add live replay eligibility checker

### LINKED ISSUE #2789: validation: add live-replay eligibility checker for diagnostic evidence
## Goal / Problem

    Add a checker that labels whether diagnostic evidence artifacts are eligible for live replay promotion.

    ## Archetype Metadata

    ```yaml
    archetype: workflow
    evidence_tier: nominal
    linked_policy:
  - docs/context/issue_1512_issue_archetypes.md
  - docs/context/artifact_evidence_vocabulary.md
  - docs/context/issue_691_benchmark_fallback_policy.md
    ```

    ## Context

    Source: attached fresh rerun over last-12-hour PR stream from 2026-06-13 14:24 CEST.

    Several artifacts are diagnostic-only because they use stored traces, smoke fixtures, or stored action proxies. The repo needs a systematic way to identify which have a live-replay path.

    ## Scope

    - Check fixture path exists, executable scenario exists, policy candidate exists, perturbation config exists, output report path is defined, live metrics are supported, and claim boundary is declared.
- Emit eligible, missing-fields, blocked, or diagnostic-only classifications.
- Do not claim benchmark evidence; only report promotion eligibility.

    ## Non-goals

    - Do not run live replay.
- Do not rewrite evidence artifacts.
- Do not promote diagnostic rows automatically.

    ## Definition of Done

    - [ ] Checker emits live-replay eligibility status with missing prerequisites.
- [ ] Fixture tests cover eligible, missing fixture, missing policy, missing metrics, and blocked claim boundary cases.
- [ ] Output can feed ledger promotion paths or issue ranking.

    ## Validation / Testing

    - [ ] Run checker tests on fixture evidence records.
- [ ] Run a dry-run against a bounded diagnostic artifact if available.
- [ ] Inspect output for fail-closed wording.

    ## Evidence Boundary

    This issue must preserve the repository's fail-closed evidence discipline. Diagnostic-only, trace-derived, stored-action-proxy, unavailable, stale, fallback, or denominator-invalid evidence must not be promoted to benchmark, dissertation, or paper claims without the named promotion proof.

## Related Evidence

Motivated by PR #2778 and open issue #2777.

    ## Project Metadata

    - Priority: High
    - Effort (h): 4-8
    - Reviewed: no

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Adds a fail-closed live-replay eligibility checker for diagnostic evidence records, plus fixture-style tests covering eligible, missing, blocked, diagnostic-only, and malformed cases.

## Linked Issues
- Closes #2789

## Stack / Dependency
- Base dependency: none
- Required prior PRs: none
- Stack follow-up issues: none
- Safe to review independently: yes
- Review dependency reason, if any: NA

## What Changed
- Added `scripts/validation/check_live_replay_eligibility.py`.
- Added `tests/validation/test_check_live_replay_eligibility.py`.
- The checker accepts JSON object/list/`evidence_records`, emits `eligible`, `missing-fields`, `blocked`, or `diagnostic-only`, and keeps the output boundary explicitly non-claim-grade.

## Why It Matters
- Added value: diagnostic artifacts can now state whether they have prerequisites for a future live-replay promotion check.
- Expected impact: improves reproducibility and ranking of diagnostic evidence without accidentally promoting it.
- Why this is worth merging now: follows #2778/#2777 evidence-boundary work and gives the repo a cheap fail-closed preflight.

## Research Result Guidance
- Target claim / hypothesis / blocker this should affect: diagnostic evidence promotion readiness only.
- Comparator or baseline, if applicable: NA
- Evidence tier: targeted smoke / support helper
- Result classification: blocker-resolution
- Decision or stop rule, if applicable: diagnostic records with missing prerequisites stay non-eligible; no benchmark/paper claim is created.
- Parent issue, claim map, registry, context note, or synthesis surface to update: #2789
- New research/benchmark/metric/paper-facing analysis tool, if any: support helper; representative dry-run used a tracked diagnostic fixture record and emitted `eligible` only for future live-replay promotion, not proof.

## Falsification / Non-Transfer Check
- Did the mechanism activate? yes, checker classified complete and incomplete fixture records.
- Did the intervention change command source, selected command, trajectory, or route progress? NA - no live replay is run.
- Did the scenario actually contain the targeted failure mode? dry-run references the occluded-emergence diagnostic fixture from #2755/#2756.
- Result route: continue to live replay only when a downstream issue runs the actual promotion proof.
- Follow-up question or issue for weak, negative, or non-transfer results: #2777 remains the live-replay proof path.

## Next Empirical Action
- Rerun needed: no for this helper.
- Extractor or analysis tool needed: no
- Artifact missing or unavailable: none for helper validation.
- Stop / revise / continue decision: stop for #2789; continue with live replay in #2777 or successor work.
- Proposed child issue or existing follow-up: #2777

## Validation / Proof
- Commands run:
  - `uv run pytest tests/validation/test_check_live_replay_eligibility.py -q`
  - `uv run pytest tests/validation/test_check_live_replay_eligibility.py tests/validation/test_check_claim_readiness.py -q`
  - `uv run ruff check scripts/validation/check_live_replay_eligibility.py tests/validation/test_check_live_replay_eligibility.py`
  - `uv run ruff format --check scripts/validation/check_live_replay_eligibility.py tests/validation/test_check_live_replay_eligibility.py`
  - `git diff --check`
  - `uv run python scripts/validation/check_live_replay_eligibility.py output/issue-2789-dry-run-record.json --base-dir . --output-json output/issue-2789-live-replay-eligibility-report.json`
  - `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main scripts/dev/pr_ready_check.sh`
- Evidence that the change works here: dry-run emitted `eligible: 1` for the tracked occluded-emergence diagnostic fixture record with an explicit not-benchmark/not-paper/not-live-proof boundary.
- Benchmarks or smoke tests, if applicable: full readiness passed with `6049 passed, 9 skipped, 9 warnings`; stamp `output/validation/pr_ready/issue-2789-live-replay-eligibility.json`.

## Risks / Rollout
- Compatibility risks: low; new standalone script and tests.
- Failure modes: input schemas may evolve; unknown/malformed records fail closed as `blocked`.
- Rollback or fallback plan: remove the new script/tests; no runtime path depends on it.

## Docs / Provenance
- Updated docs: none
- Relevant design or provenance notes: active delegation ledger records scout/implementation/review routes and dry-run output.
- Any assumptions that need to be preserved: eligibility is only a future-promotion preflight, not live replay or benchmark proof.

## Downstream Propagation
- Parent issue updated (yes/no/NA): yes, via this PR closing #2789
- Claim map / benchmark report updated (yes/no/NA): NA
- Leaderboard / artifact catalog updated (yes/no/NA): NA
- Registry or config index updated (yes/no/NA): NA
- Context index / memory note updated (yes/no/NA): NA
- Follow-up issue opened for deferred propagation (yes/no/NA): NA, existing #2777 covers live replay proof.
- Not applicable because: this helper classifies promotion eligibility but does not promote evidence.

## Follow-Up Issues
- Deferred work: actual live-replay proof remains out of scope.
- Issues opened for follow-up: none; existing #2777 remains relevant.

## Reviewer Notes
- Anything a reviewer should verify closely: CLI exit code is fail-closed; it returns zero only when every record is eligible.
- Any known limitations: the checker validates declared prerequisites and local fixture existence, not scenario executability by running the simulator.

### CODE REVIEW

File: scripts/validation/check_live_replay_eligibility.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    "output_report_path": ("output_report_path",),
+}
+BOUNDARY_MARKERS = (
+    "diagnostic",
+    "not benchmark",
+    "not paper",
+    "fail-closed",
+    "promotion",
+    "live replay",
+)
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/validation/check_live_replay_eligibility.py:35]: Fixed in 8bb38f07: boundary markers now use normalized space-separated forms.

File: scripts/validation/check_live_replay_eligibility.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+def _safe_claim_boundary(record: dict[str, Any]) -> tuple[bool, str | None]:
+    """Return whether claim boundary text is present and conservative."""
+    boundary = _text_value(record, "claim_boundary")
+    if boundary is None:
+        return False, "missing_claim_boundary"
+    lowered = boundary.lower()
+    if any(marker in lowered for marker in BOUNDARY_MARKERS):
+        return True, None
+    return False, "unsafe_claim_boundary"
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/validation/check_live_replay_eligibility.py:80]: Fixed in 8bb38f07: claim-boundary checks normalize underscores and hyphens before matching.

File: scripts/validation/check_live_replay_eligibility.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    """Classify one diagnostic evidence record."""
+    base = (base_dir or Path.cwd()).resolve()
+    if not isinstance(record, dict):
+        return {
+            "index": index,
+            "status": BLOCKED,
+            "missing_prerequisites": ["record_object"],
+            "blocked_reasons": ["malformed_record"],
+            "diagnostic_reasons": [],
+        }
```
[the remark this answers is not in the document]
Comment (ll7) [reply scripts/validation/check_live_replay_eligibility.py:125]: Fixed in 8bb38f07: malformed-record classifications now include fixture_path, policy_candidate, and claim_boundary keys set to null.
